File: test_data/csharp/test_main/aas_core_meta.v3rc2/input/snippets/Verification/reference_key_values_equal.cs
/// <summary>
/// Check that the two references, <paramref name="that" /> and
/// <paramref name="other" />, are equal by comparing
/// their <see cref="Aas.IReference.Keys" /> by
/// <see cref="Aas.IKey.Value" />'s.
/// </summary>
public static bool ReferenceKeyValuesEqual(
    Aas.IReference that,
    Aas.IReference other
)
{
    if (that.Keys.Count != other.Keys.Count)
    {
        return false;
    }

    for (int i = 0; i < that.Keys.Count; i++)
    {
        if (that.Keys[i].Value != other.Keys[i].Value)
        {
            return false;
        }
    }

    return true;
}

File: dev/test_data/main/python/expected/aas_core_meta.v3/input/snippets/Verification/reference_key_values_equal.py
def reference_key_values_equal(
    that: aas_types.Reference, other: aas_types.Reference
) -> bool:
    """
    Check that the two references, :paramref:`that` and :paramref:`other`,
    are equal by comparing their :py:attr:`.types.Reference.keys`
    by :py:attr:`.types.Key.value`'s.
    """
    if len(that.keys) != len(other.keys):
        return False

    for that_key, other_key in zip(that.keys, other.keys):
        if that_key.value != other_key.value:
            return False

    return True
